Sources/AsyncHTTPClient/RequestValidation.swift

```diff
@@ -29,28 +29,35 @@ extension HTTPHeaders {
         self.remove(name: "Transfer-Encoding")
         self.remove(name: "Content-Length")
 
-        if let body = body {
-            guard (encodings.filter { $0 == "chunked" }.count <= 1) else {
-                throw HTTPClientError.chunkedSpecifiedMultipleTimes
-            }
+        try self.validateFieldNames()
+
+        guard let body = body else {
+            // https://tools.ietf.org/html/rfc2616#section-4.3
+            //
+            // The presence of a message-body in a request is signaled by the inclusion of a
+            // Content-Length or Transfer-Encoding header field in the request's message-headers.
+            return
```

Comment:
RFC 2616 is dead, let's stop citing it. It's RFC 7230 section 3.3. _However_, this change is not cautious enough: if there's no body, but the request verb is `POST`, "Content-Length: 0" is mandatory. Content-Length: 0 is also harmless on all request types that do not define a body. A patch that users this escape hatch on only `GET`, `HEAD`, `DELETE`, `CONNECT`, and `TRACE` would be acceptable though, so long as it sends C-L 0 everywhere else.